app/partials/data-export-button.cjsx

```diff
@@ -14,15 +14,37 @@ module.exports = React.createClass
   getInitialState: ->
     exportRequested: false
     exportError: null
+    showButton: true
+    mostRecent: null
+
+  componentDidMount: ->
+    @exportGet()
+    @showDataExportBtn()
+
+  componentWillReceiveProps: ->
+    @exportGet()
+    @showDataExportBtn()
 
   exportGet: ->
-    @_exportsGet or= @props.project.get(@props.exportType).catch( -> [])
+    @props.project.get(@props.exportType)
+      .then ([exported]) =>
+        @setState mostRecent: exported
```

Comment:
`setState` is asynchronous (explanation in React docs: https://facebook.github.io/react/docs/react-component.html#setstate) so if you're calling a method that depends on the state you've just set, you can't reliably depend on it being set by the time this method is called. Three ways you can deal with this:
- call the method in `setState` as a callback. `setState` can take a callback.
- Pass `exported` in as an argument and then use it on line 39.
- Remove `showDataExportBtn` entirely and just directly add the conditional in the render `if @props.exportType isnt 'aggregations_export' or @state.mostRecent?.metadata?.state?`. When `setState` is called, it causes a re-render.